src/vs/workbench/contrib/interactiveEditor/browser/interactiveEditorStrategies.ts

```diff
@@ -257,11 +260,13 @@ export class LiveStrategy extends EditModeStrategy {
 	}
 
 	async cancel() {
-		const { textModelN: modelN, textModel0: model0 } = this._session;
-		if (modelN.isDisposed() || model0.isDisposed()) {
+		const { textModelN: modelN, textModel0: model0, lastSnapshot } = this._session;
+		if (modelN.isDisposed() || (model0.isDisposed() && !lastSnapshot)) {
 			return;
 		}
-		const edits = await this._editorWorkerService.computeMoreMinimalEdits(modelN.uri, [{ range: modelN.getFullModelRange(), text: model0.getValue() }]);
+
+		const initialTextModel = lastSnapshot ? this._modelService.createModel(createTextBufferFactoryFromSnapshot(lastSnapshot), null) : model0;
+		const edits = await this._editorWorkerService.computeMoreMinimalEdits(modelN.uri, [{ range: modelN.getFullModelRange(), text: initialTextModel.getValue() }]);
```

Comment:
We can introduce a new utility `getValueFromSnapshot()` later to avoid creating text model temporarily from the snapshot.

Reply:
Let me add such a util. Otherwise we are in the buisness of dispose this (shortlived) initialTextModel instance. The util is simpler and better